linux/RNGEntropy.c

```diff
@@ -0,0 +1,92 @@
+/*
+htop - RNGEntropy.c
+(C) 2022 Loic Jourdheuil
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "linux/RNGEntropy.h"
+
+#include <stdlib.h>
+
+#include "CRT.h"
+#include "Object.h"
+#include "XUtils.h"
+
+#define INVALID_VALUE ((unsigned int)-1)
+
+static unsigned int entropy = INVALID_VALUE;
+static unsigned int poolsize = INVALID_VALUE;
+
+static void RNGEntropy_updateValues(Meter* this) {
+   char buffer[11];
+   ssize_t r;
+
+   if (poolsize == INVALID_VALUE) {
+      r = xReadfile(PROCDIR "/sys/kernel/random/poolsize", buffer, sizeof(buffer));
+      if (r <= 0) {
+         goto err;
+      }
+      poolsize = (unsigned int)strtoul(buffer, NULL, 10);
+      if (poolsize == 0) {
+         /* invalid poolsize value */
+         poolsize = INVALID_VALUE;
```

Comment:
This will continuously check the pool size again if it fails to read this value.